lib/crimp.rb

```diff
@@ -46,4 +48,19 @@ def self.parse_hash(hash)
   def self.to_string(obj)
     "#{obj}#{obj.class}"
   end
+
+  # This is for legacy/compatibilty reason:
+  #
+  # Ruby 2.1
+  # 2.class => Fixnum
+  # Ruby > 2.4
+  # 2.class => Integer
+  #
+  # Say you have a huge number of stored keys and you migrate your app from 2.1 to >= 2.4
+  # this would cause a change of the signature for a subset of the keys which would be hard
+  # to debug especially for nested data structures.
+  #
+  def self.integer_to_string(obj)
+    "#{obj}Fixnum"
```

Comment:
Probably an edge case, but would previous invocations _always_ be dealing with `Fixnum`'s (as opposed to `Bignum`'s as well)?

Reply:
yes you're right... I'll add another _hack_  to handle this 😞 .